src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/RazorReferenceProvider.ts

```diff
@@ -0,0 +1,60 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as vscode from 'vscode';
+import { getRazorDocumentUri } from './RazorConventions';
+import { RazorLanguageFeatureBase } from './RazorLanguageFeatureBase';
+import { LanguageKind } from './RPC/LanguageKind';
+
+export class RazorReferenceProvider
+    extends RazorLanguageFeatureBase
+    implements vscode.ReferenceProvider {
+
+    public async provideReferences(
+        document: vscode.TextDocument,
+        position: vscode.Position,
+        context: vscode.ReferenceContext,
+        token: vscode.CancellationToken) {
+
+        const projection = await this.getProjection(document, position, token);
+        if (!projection) {
+            return;
+        }
+
+        const references = await vscode.commands.executeCommand<vscode.Location[]>(
+            'vscode.executeReferenceProvider',
+            projection.uri,
+            projection.position) as vscode.Location[];
+
+        const results = new Array<vscode.Location>();
+        if (projection.languageKind === LanguageKind.CSharp) {
+            for (const reference of references) {
+                const razorFile = getRazorDocumentUri(reference.uri);
+
+                // Re-map the projected hover range to the host document range
+                const res = await this.serviceClient.mapToDocumentRange(
```

Comment:
nit: res => result

Reply:
I nit'ed your nit so you could nit my nit. Nit-ception!